### PR TITLE
Add attribute to specify searchable properties

### PR DESCRIPTION
**Pull request by isak-spinit**
Need input on this one :)

We want to be able to specify which properties that should be searchable and exclude properties from being searchable. If nothing is specified, it should work as before.

### CODE REVIEW

**Review by alveflo — APPROVED** [review]

File: Spinit.CosmosDb/Analysis/DefaultAnalyzer.cs [hunk 1]
```diff
@@ -18,6 +18,8 @@ public DefaultAnalyzer()
         private static IEnumerable<ITextExtractor> CreateDefaultTextExtractors()
         {
             yield return new StringPropertyTextExtractor();
+            yield return new NumberPropertyTextExtractor();
```
Comment (martin-oom) [inline Spinit.CosmosDb/Analysis/DefaultAnalyzer.cs:21]: Documentation no longer matches implementation. And is "bool" really something we want to include per default in searches? Example: you have an entity product with a boolean property "InStock", how would this be visualized? Maybe with a checkmark icon or mayby "Yes/No", or worse translated "Ja/Nej", but to find it one must use true/false... What would happen if there are multiple boolean fields? Scenario: I want to find backpacks that are waterproof but not in stock -> "backpack true false"? So maybe boolean properties needs to have some more contextual string representation, eg needs to be application configured. Example InStock = true => "in stock" and Waterproof = true => "waterproof" so a user could search for "backpack waterproof in stock"

File: Spinit.CosmosDb/Analysis/TextExtractors/ContractResolver.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                        return base.CreateProperty(member, memberSerialization);
+                    }
+
+                    return null;
+                }
+
+                if (typeof(ICosmosEntity).IsAssignableFrom(member.DeclaringType) && member.Name == nameof(ICosmosEntity.Id))
+                    return null;
+
+                if (propertyInfo.PropertyType.IsValueType && propertyInfo.PropertyType != typeof(DateTime) && propertyInfo.PropertyType != typeof(DateTime?))
```
Comment (martin-oom) [inline Spinit.CosmosDb/Analysis/TextExtractors/ContractResolver.cs:29]: Why is value types always excluded from searches? Or am I missing something? Should it not be possible to "stringify" an enum to allow searches for it in application code?

File: Spinit.CosmosDb/Analysis/TextExtractors/BoolPropertyTextExtractor.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+﻿namespace Spinit.CosmosDb
+{
+    /// <summary>
+    /// Extracts <see cref="string"/> properties from en entity.
```
Comment (martin-oom) [inline Spinit.CosmosDb/Analysis/TextExtractors/BoolPropertyTextExtractor.cs:4]: Copy pasted documentation... should be `<see cref="bool"/>` :-)

File: Spinit.CosmosDb/Analysis/TextExtractors/NumberPropertyTextExtractor.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
+﻿namespace Spinit.CosmosDb
+{
+    /// <summary>
+    /// Extracts <see cref="string"/> properties from en entity.
```
Comment (martin-oom) [inline Spinit.CosmosDb/Analysis/TextExtractors/NumberPropertyTextExtractor.cs:4]: Copy pasted documentation...  :-)

File: Spinit.CosmosDb/Analysis/DefaultAnalyzer.cs (hunk 1 above)
Comment (isak-spinit) [reply Spinit.CosmosDb/Analysis/DefaultAnalyzer.cs:21]: It is not included by default, you need to use Searchable to include it. But maybe it does not make sense as it is done now.

File: Spinit.CosmosDb/Analysis/TextExtractors/BoolPropertyTextExtractor.cs (hunk 3 above)
Comment (isak-spinit) [reply Spinit.CosmosDb/Analysis/TextExtractors/BoolPropertyTextExtractor.cs:4]: Good catch! I will change it.

File: Spinit.CosmosDb/Analysis/TextExtractors/NumberPropertyTextExtractor.cs (hunk 4 above)
Comment (isak-spinit) [reply Spinit.CosmosDb/Analysis/TextExtractors/NumberPropertyTextExtractor.cs:4]: Good catch! I will change it.

File: Spinit.CosmosDb/Analysis/TextExtractors/ContractResolver.cs (hunk 2 above)
Comment (isak-spinit) [reply Spinit.CosmosDb/Analysis/TextExtractors/ContractResolver.cs:29]: My idea was to have it working like before if no attribute is used. The best way I found with was to have TextExtractors for each type, and to exclude them here by default. Do you have any idea to make it more clear?

**Pull request merged**